Sources/StreamChatUI/Common Views/BaseViews.swift

```diff
@@ -274,46 +195,69 @@ open class ViewController: UIViewController, AppearanceSetting, Customizable {
     }
     
     public func defaultAppearance() { /* default empty implementation */ }
-    open func setUp() { /* default empty implementation */ }
+    open func setUp() { TraitCollectionWatcher.activate() }
     open func setUpAppearance() { /* default empty implementation */ }
     open func setUpLayout() { /* default empty implementation */ }
     open func updateContent() { /* default empty implementation */ }
+    open func resetAppearance() {
+        (self as! Self).applyDefaultAppearance()
+        setUpAppearance()
+        updateContent()
+    }
+}
 
-    override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
-        super.traitCollectionDidChange(previousTraitCollection)
+/// Observes trait collection changes and calls `Customizable.resetAppearance` in child-first DFS way
+/// on view & view controller hierarchies.
+private enum TraitCollectionWatcher {
+    private class Wiretap: View {
+        var onTraitChange: () -> Void = {}
+        override func setUp() { /* do not call super to prevent loop */ } // swiftlint:disable:this overridden_super_call
+        override open func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+            super.traitCollectionDidChange(previousTraitCollection)
+
+            guard #available(iOS 12, *) else { return }
+            guard previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle else { return }
+            onTraitChange()
+        }
+    }
 
-        guard #available(iOS 12, *) else { return }
-        guard previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle else { return }
+    private static var activeTap: Wiretap!
 
-        // Execute previous pending updates
-        pendingUpdates?()
+    static func activate() {
```

Comment:
Could you document please how this function works, so it's clear from the outside?